core/src/ast/types.rs

```diff
@@ -509,6 +515,14 @@ impl TypeName {
                 TypeName::PrimitiveSlice(ltmt.clone(), *prim, StdlibOrDiplomat::Diplomat)
             }
             TypeName::Ordering => TypeName::Primitive(PrimitiveType::u8),
+            TypeName::Option(inner, _) => match **inner {
+                // Option<&T>/Option<Box<T>> are the ffi-safe way to specify options
+                TypeName::Reference(..) | TypeName::Box(..) => {
+                    TypeName::Option(inner.clone(), StdlibOrDiplomat::Stdlib)
+                }
+                // For other types (primitives, structs, enums) you should use DiplomatOption
```

Comment:
nit: this is not user documentation

```suggestion
                // For other types (primitives, structs, enums) we need DiplomatOption
```